### PR TITLE
fix contract trust FromStackItem `ContractPermissionDescriptor.Create(StackItem item)`

### PR DESCRIPTION
**Pull request by Hecate2**
https://github.com/neo-project/neo/issues/2899
https://github.com/nspcc-dev/neo-go/issues/3105
https://github.com/neo-project/neo/issues/2890
**NOT EXPECTED TO BE MERGED DIRECTLY.** Just to make `ListContracts` and `GetContract(0x12fbec62fb765ce3f55845106ccf3717716723ad)` work on Testnet T5 for now
https://github.com/Hecate2/neo/releases/tag/fix-contract-trust

### CODE REVIEW

**Comment by roman-khimov** [comment]
Seems to be legit. We have appropriate NULL stored in the DB (T5 state matches between NeoGo 0.102.0 and C# 3.6), so the only question is to read it properly.

**Review by Liaojinghui — DISMISSED** [review]

**Comment by Liaojinghui** [comment]
LGTM

File: src/Neo/SmartContract/Manifest/ContractPermissionDescriptor.cs [hunk 1]
```diff
@@ -66,6 +67,13 @@ internal ContractPermissionDescriptor(ReadOnlySpan<byte> span)
             }
         }
 
+        public static ContractPermissionDescriptor Create(StackItem item)
+        {
+            if (item == StackItem.Null)
+                return CreateWildcard();
+            return new ContractPermissionDescriptor(item.GetSpan());
```
Comment (Liaojinghui) [inline src/Neo/SmartContract/Manifest/ContractPermissionDescriptor.cs:74]:
```suggestion
            return item.Equals(StackItem.Null) ? CreateWildcard() : new ContractPermissionDescriptor(item.GetSpan());
```

File: src/Neo/SmartContract/Manifest/ContractManifest.cs [hunk 2]
```diff
@@ -81,7 +81,7 @@ void IInteroperable.FromStackItem(StackItem stackItem)
             {
                 Null _ => WildcardContainer<ContractPermissionDescriptor>.CreateWildcard(),
                 // Array array when array.Any(p => ((ByteString)p).Size == 0) => WildcardContainer<ContractPermissionDescriptor>.CreateWildcard(),
-                Array array => WildcardContainer<ContractPermissionDescriptor>.Create(array.Select(p => new ContractPermissionDescriptor(p.GetSpan())).ToArray()),
+                Array array => WildcardContainer<ContractPermissionDescriptor>.Create(array.Select(ContractPermissionDescriptor.Create).ToArray()),
```
Comment (Liaojinghui) [inline src/Neo/SmartContract/Manifest/ContractManifest.cs:84]: Well i still recommand to have this logic: if any of the permission is `any`, then the permission is `any`:

```C#
                Array array when array.Any(p => p.Equals(StackItem.Null)) => WildcardContainer<ContractPermissionDescriptor>.CreateWildcard(),
                Array array => WildcardContainer<ContractPermissionDescriptor>.Create(array.Select(ContractPermissionDescriptor.Create).ToArray()),
```

**Review by Liaojinghui — APPROVED** [review]

**Review by shargon — APPROVED** [review]

**Review by AnnaShaleva — COMMENTED** [review]
It would be nice to extend the `TestSerializeTrusts` added in https://github.com/Liaojinghui/neo/pull/1/files with a wildcard deserialisation test.

Otherwise, LGTM.

**Comment by roman-khimov** [comment]
Anything preventing the merge now? Tested, @superboyiii?

**Comment by superboyiii** [comment]
Tested OK.

**Review by superboyiii — APPROVED** [review]

**Pull request merged**